ocaml/idl/datamodel_cluster.ml

```diff
@@ -110,6 +120,33 @@ let get_network =
     ~params:[(Ref _cluster, "self", "the Cluster with the network")]
     ~lifecycle ~allowed_roles:_R_READ_ONLY ()
 
+let add_extra_network =
+  call ~name:"add_extra_network"
+    ~doc:
+      "Add network for cluster. This allows the underlying cluster stack to \
+       utilise multihoming."
+    ~params:
+      [
+        (Ref _cluster, "self", "the Cluster to which network is added")
+      ; (Int, "index", "the index of the network, representing its priority")
```

Comment:
Could add that lower means higher priority or vice versa. It's not obvious. Can you have multiple networks with the same index? Or is the index a way to identify a network. The remove call suggests the latter - so this needs to be spelled out,

Reply:
I think there is an api error `pif_index_exists_on_cluster_host`, which will be raised if the index already exists. Maybe that is not obvious enough, I will add more explanation in the doc

Reply:
I assume the indexes don't need to be continuous; so it's a unique, positive number. Adding an existing number would be an error. I assume the existing map can be observed to avoid this.

Reply:
Yes, the `extra_networks` parameter in `pool_create` has more detailed explanation about this, which is basically what you described here. And the `pif_index_exists_on_cluster_host` is raised exactly when the existing map already has that index

Reply:
I will write a design detailing all of these things, at the moment, this index seems a bit strange in how it is supposed to be used.